migration_scripts/bulk_load/src/footprint.clj

```diff
@@ -7,13 +7,324 @@
+    :objects-history/object-digest
+    :objects-history/checkpoint-sequence-number
+    :objects-history/owner-type
+    :objects-history/owner-id
+    :objects-history/object-type
+    :objects-history/object-type-package
+    :objects-history/object-type-module
+    :objects-history/object-type-name
+    :objects-history/serialized-object
+    :objects-history/coin-type
+    :objects-history/coin-balance
+    :objects-history/df-kind
+    :objects-history/df-name
+    :objects-history/df-object-type
+    :objects-history/df-object-id
+
+    :objects-snapshot/object-id
+    :objects-snapshot/object-version
+    :objects-snapshot/object-status
+    :objects-snapshot/object-digest
+    :objects-snapshot/checkpoint-sequence-number
+    :objects-snapshot/owner-type
+    :objects-snapshot/owner-id
+    :objects-snapshot/object-type
+    :objects-snapshot/object-type-package
+    :objects-snapshot/object-type-module
+    :objects-snapshot/object-type-name
+    :objects-snapshot/serialized-object
+    :objects-snapshot/coin-type
+    :objects-snapshot/coin-balance
+    :objects-snapshot/df-kind
+    :objects-snapshot/df-name
+    :objects-snapshot/df-object-type
+    :objects-snapshot/df-object-id
+
+    :objects-version/object-id
+    :objects-version/object-version
+    :objects-version/cp-sequence-number
+
+    :packages/package-id
+    :packages/original-id
+    :packages/package-version
+    :packages/move-package
+    :packages/checkpoint-sequence-number
+
+    :transactions/tx-sequence-number
+    :transactions/transaction-digest
+    :transactions/raw-transaction
+    :transactions/raw-effects
+    :transactions/checkpoint-sequence-number
+    :transactions/timestamp-ms
+    :transactions/object-changes
+    :transactions/balance-changes
+    :transactions/events
+    :transactions/transaction-kind
+    :transactions/success-command-count
+
+    :tx-calls-pkg/tx-sequence-number
+    :tx-calls-pkg/package
+    :tx-calls-pkg/sender
+
+    :tx-calls-mod/tx-sequence-number
+    :tx-calls-mod/package
+    :tx-calls-mod/module
+    :tx-calls-mod/sender
+
+    :tx-calls-fun/tx-sequence-number
+    :tx-calls-fun/package
+    :tx-calls-fun/module
+    :tx-calls-fun/func
+    :tx-calls-fun/sender
+
+    :tx-changed-objects/tx-sequence-number
+    :tx-changed-objects/object-id
+    :tx-changed-objects/sender
+
+    :tx-digests/tx-digest
+    :tx-digests/tx-sequence-number
+
+    :tx-input-objects/tx-sequence-number
+    :tx-input-objects/object-id
+    :tx-input-objects/sender
+
+    :tx-kinds/tx-sequence-number
+    :tx-kinds/tx-kind
+
+    :tx-recipients/tx-sequence-number
+    :tx-recipients/recipient
+    :tx-recipients/sender
+
+    :tx-senders/tx-sequence-number
+    :tx-senders/sender})
+
+
+(def kv-columns
+  "Columns that could be moved to a blob store.
+
+  Columns whose names end with a `*` can be omitted with the
+  introduction of a blob store. Those that end in a `+` must be
+  duplicated, and plain columns are moved."
```

Comment:
it seems like either a column can be dropped with a kv store, or it must be duplicated on both the kv and postgres tables. wouldn't the plain columsn then also be `+`d?

Reply:
There are four states that a column in the original schema can be in, with respect to the KV store:

- Left alone (I think that's what you are describing as a "plain column") -- such columns just don't appear in this list.
- Added to the KV store and removed from postgres (this is what I am referring to as a "plain column" above, because its representation in this set is a name without any suffix).
- Added to the KV store and retained in postgres (end in `+`).
- Removed from postgres but not added to the KV store (end in `*`).

Reply:
got it, i conflated some of the states together